src/uct/ib/ud/base/ud_iface.h

```diff
@@ -177,6 +245,12 @@ struct uct_ud_iface {
         uct_async_event_cb_t      event_cb;
         unsigned                  disable;
     } async;
+
+    /* used for GRH GID filter */
+    struct {
+        khash_t(uct_ud_iface_gid) hash;
+        union ibv_gid             last;
```

Comment:
so we don't need len, because we expand ipv4 gid with 0-s anyway?
IMO we can optimize it (maybe next PR)
BTW did you see if this PR affects UD performance on RoCE (osu_lat, osu_bw)?

Reply:
did not check in current version but previous one has the same logic starting from 2nd packet.
will retest.